solr/core/src/java/org/apache/solr/cloud/overseer/ZkStateWriter.java

```diff
@@ -301,6 +311,8 @@ public ClusterState writePendingUpdates(Map<String, ZkWriteCommand> updates, boo
       // this is a tragic error, we must disallow usage of this instance
       invalidState = true;
       throw bve;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Comment:
assuming `CompressionUtil.compressBytes()` should not throw `IOException`, this catch clause should be removed as well.